xwiki-platform-core/xwiki-platform-skin/xwiki-platform-skin-skinx/src/main/java/com/xpn/xwiki/plugin/skinx/AbstractDocumentSkinExtensionPlugin.java

```diff
@@ -454,9 +467,10 @@ private String getLanguageQueryString(XWikiContext context)
     protected String getDocumentSkinExtensionURL(DocumentReference documentReference, String documentName,
             String pluginName, XWikiContext context)
     {
-        String queryString = String.format("%s&amp;%s%s",
+        String queryString = String.format("%s&amp;%s&amp;%s%s",
                 getLanguageQueryString(context),
                 getDocumentVersionQueryString(documentReference, context),
+                getWikiVersionQueryString(context),
```

Comment:
I don't see the point for a wiki page, if the wiki page did not changed then why could we need to invalidate the cache ?

Reply:
Here it's really for a dev usecase: specifically if I launch on my own machine XWiki 10.8.2 and then I launch XWiki 10.11, the doc version won't change so it will try to get it from the cache.
I know this shouldn't happen in prod since most of the case it would be the result of a migration and the doc version should change.
We can still imagine the usecase of an user willing to upgrade its version by installing a fresh XWiki, so here it might be useful for this specific case.

Reply:
Hmm maybe a better criteria for this use case is the instance id then.